src/helpViewer/helpProvider.ts

```diff
@@ -270,33 +268,42 @@ export class AliasProvider {
     }
 
     // call R script `getAliases.R` and parse the output
-    private getAliasesFromR(): undefined | AllPackageAliases {
-
-        // get from R
-		const lim = '---vsc---'; // must match the lim used in R!
-		const re = new RegExp(`^.*?${lim}(.*)${lim}.*$`, 'ms');
-        const tempDir = fs.mkdtempSync(path.join(os.tmpdir(), 'vscode-R-aliases'));
-        const tempFile = path.join(tempDir, 'aliases.json');
-        const cmd = `"${this.rPath}" --silent --no-save --no-restore --slave -f "${this.rScriptFile}" > "${tempFile}"`;
-
-        let allPackageAliases: undefined | AllPackageAliases = undefined;
-        try{
-            // execute R script 'getAliases.R'
-            // aliases will be written to tempFile
-            cp.execSync(cmd, { cwd: this.cwd });
-
-            // read and parse aliases
-            const txt = fs.readFileSync(tempFile, 'utf-8');
-            const json = txt.replace(re, '$1');
-            if(json){
-                allPackageAliases = <{[key: string]: PackageAliases}> JSON.parse(json) || {};
+    private async getAliasesFromR(): Promise<undefined | AllPackageAliases> {
+        const lim = '---vsc---';
+        const options: cp.CommonOptions = {
+            cwd: this.cwd,
+            env: {
+                ...process.env,
+                VSCR_LIM: lim
             }
-        } catch(e: unknown){
+        };
+
+        const args = [
+            '--silent',
+            '--slave',
+            '--no-save',
+            '--no-restore',
+            '-f',
+            this.rScriptFile
+        ];
+        
+        try {
+            const result = await spawnAsync(this.rPath, args, options);
+            if (result.status === 0) {
+                const re = new RegExp(`${lim}(.*)${lim}`, 'ms');
+                const match = re.exec(result.stdout);
```

Comment:
Just a stylistic suggestion to reduce nesting/improve readability:
``` ts
            const result = await spawnAsync(this.rPath, args, options);
            if (result.status !== 0) {
                throw result.error || new Error(result.stderr);
            }
            const re = new RegExp(`${lim}(.*)${lim}`, 'ms');
            const match = re.exec(result.stdout);
            if(match.length !== 2){
                throw new Error('Could not parse R output.');
            }
            const json = match[1];
            return <AllPackageAliases>JSON.parse(json) || {};
```

Reply:
Good suggestion. Looks better now.